runtime/tests/integration_tests.rs

```diff
@@ -425,6 +430,148 @@ fn test_invoke_dpe_get_certificate_chain_cmd() {
     assert_ne!([0u8; 2048], cert_chain.certificate_chain);
 }
 
+#[test]
+fn test_pauser_privilege_level_dpe_context_thresholds() {
+    let mut model = run_rt_test(None, None);
+
+    model.step_until(|m| {
+        m.soc_ifc().cptra_boot_status().read() == RtBootStatus::RtReadyForCommands.into()
+    });
+
+    let mut cmd_data: [u8; 512] = [0u8; InvokeDpeReq::DATA_MAX_SIZE];
+    // First rotate the default context so that we don't run into an error
+    // when trying to retain the default context in derive child.
+    let rotate_ctx_cmd = RotateCtxCmd {
+        handle: ContextHandle::default(),
+        flags: RotateCtxFlags::empty(),
+    };
+    let rotate_ctx_cmd_hdr = CommandHdr::new_for_test(Command::ROTATE_CONTEXT_HANDLE);
+    let rotate_ctx_cmd_hdr_buf = rotate_ctx_cmd_hdr.as_bytes();
+    cmd_data[..rotate_ctx_cmd_hdr_buf.len()].copy_from_slice(rotate_ctx_cmd_hdr_buf);
+    let rotate_ctx_cmd_buf = rotate_ctx_cmd.as_bytes();
+    cmd_data[rotate_ctx_cmd_hdr_buf.len()..rotate_ctx_cmd_hdr_buf.len() + rotate_ctx_cmd_buf.len()]
+        .copy_from_slice(rotate_ctx_cmd_buf);
+    let rotate_ctx_mbox_cmd = InvokeDpeReq {
+        hdr: MailboxReqHeader { chksum: 0 },
+        data: cmd_data,
+        data_size: (rotate_ctx_cmd_hdr_buf.len() + rotate_ctx_cmd_buf.len()) as u32,
+    };
+
+    let checksum = caliptra_common::checksum::calc_checksum(
+        u32::from(CommandId::INVOKE_DPE),
+        &rotate_ctx_mbox_cmd.as_bytes()[4..],
+    );
+
+    let rotate_ctx_mbox_cmd = InvokeDpeReq {
+        hdr: MailboxReqHeader { chksum: checksum },
+        ..rotate_ctx_mbox_cmd
+    };
+
+    let rotate_ctx_resp_buf = model
+        .mailbox_execute(
+            u32::from(CommandId::INVOKE_DPE),
+            rotate_ctx_mbox_cmd.as_bytes(),
+        )
+        .unwrap()
+        .expect("We should have received a response");
+
+    assert!(rotate_ctx_resp_buf.len() <= std::mem::size_of::<InvokeDpeResp>());
+    let mut rotate_ctx_resp_hdr = InvokeDpeResp::default();
+    rotate_ctx_resp_hdr.as_bytes_mut()[..rotate_ctx_resp_buf.len()]
+        .copy_from_slice(&rotate_ctx_resp_buf);
```

Comment:
nit: I think these lines can collapse to `let rotate_ctx_resp = InvokeDpeResp::read_from(rotate_ctx_resp_buf)`

Reply:
I tried it and the read_from call fails. Not totally sure, but I think it is due to missing the MailboxRespHeader.

Reply:
Does it work if you do `let rotate_ctx_resp = InvokeDpeResp::read_from_prefix(rotate_ctx_resp_buf))`? I'm guessing it fails because the slice it larger than `read_from` is expecting.

Reply:
No, strangely it still fails but now it fails on the call to DeriveChild. I'm guessing it incorrectly parsed the rotated context handle so derive child couldn't find it.